src/util/cilfacade.ml

```diff
@@ -79,6 +79,81 @@ let end_basic_blocks f =
   let thisVisitor = new allBBVisitor in
   visitCilFileSameGlobals thisVisitor f
 
+class loopUnrollingVisitor = object
+  inherit nopCilVisitor
+  method! vstmt s =
+    (* All the duplicated stmts need to be re-created or the sid will fail*)
+    let newsid st = { st with sid = st.sid } in
+    let rec newsid_stmt st = 
+      let mkb stml = mkBlock (List.map newsid_stmt stml) in
+      match st.skind with
+      | Instr _ -> newsid st
+      | If(e,b1,b2,l1,l2) -> mkStmt(If(e,(mkb b1.bstmts),(mkb b2.bstmts),l1,l2))
+      | Loop(bl,l1,l2,s1,s2) -> 
+        let create_stmt = mkStmt(Loop((mkb bl.bstmts),l1,l2,s1,s2)) in
+        create_stmt.labels <- st.labels;
+        create_stmt
+      | Block(bl) -> mkStmt(Block(mkb bl.bstmts))
+      | Switch(e,bl,stl,l1,l2) -> mkStmt(Switch(e,(mkb bl.bstmts),(List.map newsid_stmt stl),l1,l2))
+      | _ -> newsid st in
+    let mkBlock' b = mkBlock (List.map newsid_stmt b) in
+    let mk_stmt_from_stmt_list bl = mkStmt(Block(mkBlock' bl)) in
+    match s.skind with
+    | Loop(b, loc, _, _, _) ->
+      let get_unrolling_factor = GobConfig.get_int "exp.unrolling-factor" in
+      (* All unrollings will leave the original loop at the end. The label makes sure it's not unrolled twice.*)
+      let add_label_to_remainder_loop st_loop loc_loop = 
+        st_loop.labels <- [Label(Cil.freshLabel "remainder_loop", loc_loop, false)];
+        st_loop in
+      let is_remainder_loop loop_stmt_labels =
+        match loop_stmt_labels with
+        | [] -> false
+        | Label(lab,_,_)::tl ->
+          begin
+            try
+              match String.sub lab 0 14 with
+              | "remainder_loop" -> true
+              | _ -> false
+            with _-> false
+          end
```

Comment:
```suggestion
          BatString.starts_with lab "remainder_loop"
```

Reply:
Also, this doesn't recurse into `tl`, so it just checks the first label. I guess this could be `List.exists` instead.